app/domain/authentication/authn_jwt/signing_key/create_signing_key_interface.rb

```diff
@@ -0,0 +1,75 @@
+          if provider_uri_has_valid_configuration?
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedSigningKeyInterface.new(PROVIDER_URI_INTERFACE_NAME))
+            fetch_provider_uri_signing_key
+          elsif jwks_uri_has_valid_configuration?
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedSigningKeyInterface.new(JWKS_URI_INTERFACE_NAME))
+            fetch_jwks_uri_signing_key
+          end
+        end
+
+        def validate_key_configuration
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatingJwtSigningKeyConfiguration.new)
+          if (provider_uri_has_valid_configuration? && jwks_uri_has_valid_configuration?) ||
+              (!provider_uri_has_valid_configuration? && !jwks_uri_has_valid_configuration?)
+            raise Errors::Authentication::AuthnJwt::InvalidUriConfiguration.new(
+              PROVIDER_URI_RESOURCE_NAME,
+              JWKS_URI_RESOURCE_NAME
+            )
+          end
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedSigningKeyConfiguration.new)
+        end
+
+        def provider_uri_has_valid_configuration?
+          return @provider_uri_has_valid_configuration unless @provider_uri_has_valid_configuration.nil?
+          @provider_uri_has_valid_configuration ||= fetch_provider_uri_signing_key.valid_configuration?
+        end
+
+        def jwks_uri_has_valid_configuration?
+          return @jwks_uri_has_valid_configuration unless @jwks_uri_has_valid_configuration.nil?
+          @jwks_uri_has_valid_configuration ||= fetch_jwks_uri_signing_key.valid_configuration?
+        end
+
+        def fetch_provider_uri_signing_key
+          @fetch_provider_uri_signing_key ||= @fetch_provider_uri.new(authentication_parameters: @authentication_parameters,
+                                                                      logger: Rails.logger,
+                                                                      fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+                                                                      resource_class: ::Resource,
+                                                                      discover_identity_provider: Authentication::OAuth::DiscoverIdentityProvider.new)
+        end
+
+        def fetch_jwks_uri_signing_key
+          @fetch_jwks_uri_signing_key ||= @fetch_jwks_uri.new(authentication_parameters: @authentication_parameters,
+                                                              logger: Rails.logger,
+                                                              fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+                                                              resource_class: ::Resource,
+                                                              http_lib: Net::HTTP,
+                                                              create_jwks_from_http_response: CreateJwksFromHttpResponse.new)
```

Comment:
move all these to dependencies of `fetch_jwks_uri`

Reply:
I agree with Tamir, this is the most convenient way to "simulate" dependency for those regular classes so we cant test them with UTs easy

Reply:
1. not sure why they are not command classes but they do sound like ones.
2. anyway, in regular classes we set dependencies in the method signature with default values. there's no value in passing `http_lib: Net::HTTP` (for example) in the code itself from the calling class. this should be set in the init method signature so that app code does not bother with this and UTs can pass mocks.

Reply:
Because we call it not one time but three times. One from factory one to get public key and one from status API.
I will move depencencies to constructor